userspace/libsinsp/test/events_proc.ut.cpp

```diff
@@ -355,6 +356,14 @@ TEST_F(sinsp_with_test_input, spawn_process)
 	ASSERT_EQ(get_field_as_string(evt, "proc.sname"), "init");
 	ASSERT_EQ(get_field_as_string(evt, "proc.sid.exe"), "/sbin/init");
 	ASSERT_EQ(get_field_as_string(evt, "proc.sid.exepath"), "/sbin/init");
+	ASSERT_EQ(get_field_as_string(evt, "proc.is_sid_leader"), "false");
```

Comment:
Of course, I appreciate all the tests but I think we need to go a step further in what concern the testing. If we put ourselves in the simple case we won't discover possible failures, we probably need a real tree on which we can assert these values. Here i proposed a first attempt https://github.com/falcosecurity/libs/pull/1182/files#diff-505fbdbaff39fa9ce959ca0c186a07c3aaa89c4c3850b5d7913e74edcfab923a of course the tree is still simple but at least we have a namespace and we can try to assert something

Reply:
this is not related to this PR in particular but it is more of a generic comment, we probably need a generic process tree with 2/3 namespaces on which we assert all these process id values

Reply:
I would say after the other PR introduce them :+1: but yes we can start thinking about how to expand them. 
Right now they are mainly focused on 3 fields: tid, pid, ptid, but they should be extensible to other fields like vtid, vpid, vpgid, ecc... 
I have 2 main ideas in mind:
1.  Enrich the DEFAULT TREE with other processes and maybe another namespace and allow assertion on many different fields
2. Produce scap-fileS (:rofl: ) and parse them matching the actual system with a pstree snapshot as I've done here https://github.com/falcosecurity/libs/pull/1182/files#diff-e62ccb1906fa6f9ad858340e2d9b177b413a55e4a2dd573527265d3c8b84e812

(1) should help us in asserting that if the driver sends correct data userspace behaves well
(2) should help in understanding if our sinsp state reflects the running system

Reply:
I am curious now :smile: Meanwhile here some good old JSON lines of me running a py script in a container.

```
What I typed and executed from my terminal in a container

root@05df91a3e423:/test# whoami
root@05df91a3e423:/test# python3 memfd.p

{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"whoami","proc.exepath":"/usr/bin/whoami","proc.is_sid_leader":false,"proc.is_vpgid_leader":true,"proc.pid":947514,"proc.ppid":901354,"proc.pvpid":1,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":847,"proc.vpgid.exe":"whoami","proc.vpgid.exepath":"/usr/bin/whoami","proc.vpgid.name":"whoami","proc.vpid":847}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"python3 memfd.py","proc.exepath":"/usr/bin/python3","proc.is_sid_leader":false,"proc.is_vpgid_leader":true,"proc.pid":947555,"proc.ppid":901354,"proc.pvpid":1,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":848}

{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"ldconfig /sbin/ldconfig -p","proc.exepath":"/sbin/ldconfig","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947556,"proc.ppid":947555,"proc.pvpid":848,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":849}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"ldconfig.real -p","proc.exepath":"/sbin/ldconfig.real","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947556,"proc.ppid":947555,"proc.pvpid":848,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":849}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"sh -c bash -c 'exec ls -l /proc/848/fd/'","proc.exepath":"/bin/sh","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947557,"proc.ppid":947555,"proc.pvpid":848,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":850}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"bash -c exec ls -l /proc/848/fd/","proc.exepath":"/usr/bin/bash","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947558,"proc.ppid":947557,"proc.pvpid":850,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":851}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"ls -l /proc/848/fd/","proc.exepath":"/usr/bin/ls","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947558,"proc.ppid":947557,"proc.pvpid":850,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":851}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"sh -c bash -c 'exec /proc/848/fd/3 -l /proc/848/fd/'","proc.exepath":"/bin/sh","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947559,"proc.ppid":947555,"proc.pvpid":848,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":852}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"bash -c exec /proc/848/fd/3 -l /proc/848/fd/","proc.exepath":"/usr/bin/bash","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947560,"proc.ppid":947559,"proc.pvpid":852,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":853}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"3 -l /proc/848/fd/","proc.exepath":"/proc/848/fd/3","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947560,"proc.ppid":947559,"proc.pvpid":852,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":853}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"sh -c whoami","proc.exepath":"/bin/sh","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947561,"proc.ppid":947555,"proc.pvpid":848,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":854}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"whoami","proc.exepath":"/usr/bin/whoami","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947562,"proc.ppid":947561,"proc.pvpid":854,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":855}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"sh -c bash -c 'echo 123456789 | base64'","proc.exepath":"/bin/sh","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947563,"proc.ppid":947555,"proc.pvpid":848,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":856}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"bash -c echo 123456789 | base64","proc.exepath":"/usr/bin/bash","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947564,"proc.ppid":947563,"proc.pvpid":856,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":857}
{"container.id":"05df91a3e423","evt.type":"execve","proc.cmdline":"base64","proc.exepath":"/usr/bin/base64","proc.is_sid_leader":false,"proc.is_vpgid_leader":false,"proc.pid":947566,"proc.ppid":947564,"proc.pvpid":857,"proc.sid":901354,"proc.sid.exepath":"/usr/bin/bash","proc.sname":"bash","proc.tty":34816,"proc.vpgid":848,"proc.vpgid.exe":"python3","proc.vpgid.exepath":"/usr/bin/python3","proc.vpgid.name":"python3","proc.vpid":859}

```